### PR TITLE
Add AMD instance types for greater capacity, lower costs

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

- AMD instance types are not used for Batch

Issue Number: N/A

## What is the new behavior?

- AMD instance types (which are cheaper, and add capacity) can be used for Batch

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

To test I ran a bunch of jobs through Batch, and verified that at least some of them used AMD instances. There were no errors.

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -167,6 +167,7 @@ module "batch_data_lake_writer" {
   use_ephemeral_storage    = true
   # SPOT is actually the default, this is just a placeholder until GTC-1791 is done
   launch_type              = "SPOT"
+  instance_types           = ["r5d.4xlarge", "r5d.8xlarge", "r5d.12xlarge", "r5d.16xlarge", "r5d.24xlarge", "r5ad.4xlarge", "r5ad.8xlarge", "r5ad.12xlarge", "r5ad.16xlarge", "r5ad.24xlarge", "c5d.12xlarge", "c5d.18xlarge", "c5d.24xlarge"]
```
Comment (gtempus) [inline terraform/main.tf:170]: 👋 The AMD instances are `r5ad`, right? The others are still intel, @dmannarino?

**Review by gtempus — APPROVED** [review]
I see this is new to `main.tf`. How were we picking instance types before, @dmannarino?

File: terraform/main.tf (hunk 1 above)
Comment (dmannarino) [reply terraform/main.tf:170]: Exactly.

**Review by jterry64 — APPROVED** [review]
Looks good, although could add some info to the PR description on the testing you did to confirm AMD works, for future reference?

**Comment by dmannarino** [comment]
> I see this is new to `main.tf`. How were we picking instance types before, @dmannarino?
Good question! They are specified as defaults (the same as here, minus the AMD types) in the wri/gfw-terraform-modules repo. See here: https://github.com/wri/gfw-data-api/blob/6ad9e6d2820ea2a254502f5013e6d2b4c985fd41/terraform/main.tf#L146-L147

**Review by jterry64 — APPROVED** [review]

**Pull request merged**